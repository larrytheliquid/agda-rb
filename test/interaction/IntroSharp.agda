-- The "intro" command manages to refine goals of type ∞ A with the
-- term ♯ ?.

{-# OPTIONS --universe-polymorphism #-}

module IntroSharp where

postulate
  Level : Set
  zero : Level
  suc  : (i : Level) → Level
  _⊔_ : Level -> Level -> Level

{-# BUILTIN LEVEL     Level #-}
{-# BUILTIN LEVELZERO zero  #-}
{-# BUILTIN LEVELSUC  suc   #-}
{-# BUILTIN LEVELMAX  _⊔_   #-}

postulate
  ∞  : ∀ {a} (A : Set a) → Set a
  ♯_ : ∀ {a} {A : Set a} → A → ∞ A
  ♭  : ∀ {a} {A : Set a} → ∞ A → A

{-# BUILTIN INFINITY ∞  #-}
{-# BUILTIN SHARP    ♯_ #-}
{-# BUILTIN FLAT     ♭  #-}

Foo : ∞ Set
Foo = ?
