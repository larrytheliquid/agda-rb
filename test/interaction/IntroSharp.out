agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action "*All Goals*" "?0 : ∞ Set
")
agda2_mode_code (agda2-goals-action '(0))
agda2_mode_code (agda2-give-action 0 "♯ ?")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action "*All Goals*" "?1 : Set
")
agda2_mode_code (agda2-goals-action '(1))
